ctru-rs/src/applets/error.rs

```diff
@@ -94,9 +94,41 @@ impl PopUp {
     }
 }
 
+struct ErrorConfWriter<'a> {
+    error_conf: &'a mut errorConf,
+    index: usize,
+}
+
+impl std::fmt::Write for ErrorConfWriter<'_> {
+    fn write_str(&mut self, s: &str) -> Result<(), std::fmt::Error> {
+        let max = self.error_conf.Text.len() - 1;
+
+        for code_unit in s.encode_utf16() {
+            if self.index == max {
+                self.error_conf.Text[self.index] = 0;
+                return Err(std::fmt::Error);
+            } else {
+                self.error_conf.Text[self.index] = code_unit;
+                self.index += 1;
+            }
+        }
+
+        self.error_conf.Text[self.index] = 0;
+
+        Ok(())
+    }
+}
+
 pub(crate) fn set_panic_hook(call_old_hook: bool) {
     use crate::services::gfx::GFX_ACTIVE;
-    use std::sync::TryLockError;
+    use std::fmt::Write;
+    use std::sync::{Mutex, TryLockError};
+
+    static ERROR_CONF: Mutex<errorConf> = unsafe { Mutex::new(std::mem::zeroed()) };
```

Comment:
imo this static should be exposed somewhere so other users of ctru-rs can use it?